fix_task_type.py

```diff
@@ -9,99 +9,222 @@
-                   "minimum_should_match" : 1,
-                 }
+body = {
+    "query": {
+        "bool": {
+            "filter": [
+                {"exists": {"field": "TaskType"}},
+                {"exists": {"field": "WMAgent_SubTaskName"}},
+                {
+                    "bool": {
+                        "should": [
+                            {
+                                "bool": {
+                                    "must_not": {
+                                        "exists": {"field": "HasTaskTypeCorrectionV1"}
+                                    }
+                                }
+                            },
+                            {"term": {"TaskType": "steponeproc"}},
+                            {"term": {"TaskType": "steptwoproc"}},
+                            {"term": {"TaskType": "unknown"}},
+                            {"term": {"TaskType": "production"}},
+                        ],
+                        "minimum_should_match": 1,
+                    }
+                },
+            ]
         }
-      ]
     }
-  }
 }
 body = json.dumps(body)
 now = int(time.time())
 
+
 def task_type_logic(result):
     workflow = result.get("Workflow", "/UNKNOWN")
-    if re.match('amaltaro_.+_Test_Backfill', workflow) or re.match('amaltaro_TEST-Backfill', workflow):
-        result['TaskType'] = 'Backfill'
-        result['Campaign'] = 'Backfill'
+    if re.match("amaltaro_.+_Test_Backfill", workflow) or re.match(
+        "amaltaro_TEST-Backfill", workflow
+    ):
+        result["TaskType"] = "Backfill"
+        result["Campaign"] = "Backfill"
         return result
-    if re.match('amaltaro_.+_HG1[0-9]+.*Val', workflow) or \
-       re.match('amaltaro_.+_Tests', workflow) or re.match('amaltaro_.+_Agent.+_Validation', workflow) or \
-       re.match('sryu_.+_reqmgr2', workflow) or re.match('amaltaro_.+_JDLMagic', workflow) or \
-       re.match('amaltaro_MonteCarlo_eff_December_Patches', workflow) or re.match('amaltaro_.+_DMWM_Test', workflow) or \
-       re.match('amaltaro_MonteCarlo_eff', workflow) or re.match('amaltaro_.+New_JSONs', workflow) or \
-       re.match('amaltaro_MonteCarlo_.+_JDL_Magic', workflow) or re.match('amaltaro_.+_December_Patches', workflow) or \
-       re.match('prozober_.+_CSCS_HPC__Commissioning', workflow) or re.match('amaltaro_.+_December_Abort', workflow) or \
-       re.match('prozober_.+_HIP_Commissioning', workflow) or re.match('prozober_.+_JAN_Test', workflow) or \
-       re.match('prozober_.+_Brunel_Commissioning', workflow) or re.match('prozober_.+_HPC_Test', workflow) or \
-       re.match('amaltaro_.+_June_Validation', workflow) or re.match('amaltaro_.+_May_Validation', workflow) or \
-       re.match('amaltaro_.+_July_Validation', workflow) or \
-       re.match('amaltaro_.+_August_Validation', workflow):
-        result['TaskType'] = 'Validation'
-        result['Campaign'] = 'Validation'
+    if (
```

Comment:
I think this script was written to be used one specific time (is covered by this PR only because it was automatically converted). Should we keep it in the current version or should we remove it?

AFAIK, the spider doesn't use this script.

Reply:
The general rule if it does not use it, remove it. The github history keeps track of changes anyway and if something needs to be recored it can be done.